### PR TITLE
Migrate Node4 lambdas to Node8

### PR DESCRIPTION
**Pull request by rahilb**
Node4 lambdas are end of lifed/deprecated. this migrates all node 4 lambdas to node 8.

### CODE REVIEW

File: node-lambdas/src/main/nodejs/startAccountDeletion.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+'use strict';
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/startAccountDeletion.js]: 👍

File: node-lambdas/src/main/nodejs/startAccountDeletion.js [hunk 2]
```diff
@@ -30,31 +31,31 @@
...
-var stepfunctions = new AWS.StepFunctions();
-var kms = new AWS.KMS();
+const AWS = require('aws-sdk');
+const stepfunctions = new AWS.StepFunctions();
+const kms = new AWS.KMS();
 const crypto = require("crypto");
 
 
-exports.handler = (event, context, callback) => {
+function startAccountDeletion(event) {
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/startAccountDeletion.js]: could be `const startAccountDeletion = event => {`

**Comment by rahilb** [comment]
> wrap the request-to-promise thing

I'll do it 👍

File: node-lambdas/package.json [hunk 3]
```diff
@@ -11,5 +11,9 @@
   "scripts": {
     "test": "mocha \"src/test/nodejs/*.js\"",
     "watch": "mocha -w \"src/test/nodejs/*.js\""
+  },
+  "dependencies": {
+    "request": "^2.87.0",
```
Comment (walaura) [inline node-lambdas/package.json]: do we still need `request`?

File: node-lambdas/src/main/nodejs/deleteUser.js [hunk 4]
```diff
@@ -1,57 +1,38 @@
...
+            'X-GU-ID-Client-Access-Token': decryptedInput.clientAccessToken,
+            'x-api-key': decryptedInput.xApiKey,
+            'Referer': 'https://' + process.env.DOTCOM_ROOT,
+            'Content-Type': "application/json",
+        },
+        method: 'DELETE',
+        json: true,
+        transform2xxOnly: true
+    }).then((json) => {
+        return json
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/deleteUser.js]: this bit is redundant, you can just have `catch`

File: node-lambdas/src/main/nodejs/userIsNotMember.js [hunk 5]
```diff
@@ -1,42 +1,22 @@
...
-
-        const request = http.request(options);
-
-        request.on('error', networkError => reject(networkError));
-
-        request.on('response', response => processResponse(response));
-
-        request.end();
+    }).catch((errorResponse) => {
+        throw new Error(`Unexpected Response Status Received: ${response.statusCode}`)
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/userIsNotMember.js]: `response` is undefined

File: node-lambdas/package.json (hunk 3 above)
Comment (rahilb) [reply node-lambdas/package.json]: Yes, it's a peer dependency of `request-promise-native`, without it NPM warns us: 

```
request-promise-core@1.1.1 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.
```

File: node-lambdas/src/main/nodejs/publishDeletion.js [hunk 6]
```diff
@@ -23,11 +23,12 @@ function decrypt(blob) {
...
+exports.handler = async (event, context) => {
     const blob = event.credentials.stateMachineInput.CiphertextBlob;
 
-    decrypt(blob)
+    return await decrypt(blob)
         .then((input) => publish(process.env.TOPIC_ARN, input))
-        .then((result) => callback(null, true))
-        .catch((error) => callback(error));
+        .then((result) => {
+            return true
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/publishDeletion.js]: do we explicitly need `true`?

File: node-lambdas/src/main/nodejs/startAccountDeletion.js [hunk 7]
```diff
@@ -89,10 +86,10 @@ exports.handler = (event, context, callback) => {
                                 if (executionResult.status === "RUNNING")
                                     setTimeout(checkExecutionFinished(executionArn), 5000);
                                 else {
-                                    var responseCode = (executionResult.status === "SUCCEEDED") ? 200 : 500;
-                                    var responseMessage = `Account deletion ${executionResult.status} for user ${identityId}`;
+                                    const responseCode = (executionResult.status === "SUCCEEDED") ? 200 : 500;
+                                    const responseMessage = `Account deletion ${executionResult.status} for user ${identityId}`;
 
-                                    var responseBody;
+                                    let responseBody;
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/startAccountDeletion.js]: do we need this anymore? you could just return directly lower down the line

File: node-lambdas/src/main/nodejs/updateLastActivityTimestamp.js [hunk 8]
```diff
@@ -4,33 +4,20 @@
...
+        uri: `https://${process.env.IDAPI_ROOT}/user/${input.userId}/activity`,
+        headers: {
+            'X-GU-ID-Client-Access-Token': ['Bearer', process.env.CLIENT_ACCESS_TOKEN].join(' ')
+        },
+        method: 'POST',
+        transform2xxOnly: true
+    }).then(() => {
+        return input;
+    }).catch((errorResponse) => {
+        throw new Error(`Non 200 response receieved ${errorResponse.statusCode}`)
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/updateLastActivityTimestamp.js]: typo

**Comment by rahilb** [comment]
>nitpicky comments

I D I O M A T I C

**Pull request merged**